src/integer_value.cpp

```diff
@@ -116,27 +117,26 @@ ValuePtr IntegerValue::mul(Env *env, ValuePtr arg) {
     }
     arg.assert_type(env, Value::Type::Integer, "Integer");
 
+    if (is_zero() || arg->as_integer()->is_zero())
+        return new IntegerValue { 0 };
```

Comment:
```suggestion
        return ValuePtr::integer(0);
```

This will avoid a heap allocation.

Reply:
Well, you either amended the commit, or I had a really outdated page in my browser. This looks resolved.